bookshelf/app/models/book.rb

```diff
@@ -0,0 +1,168 @@
+# Copyright 2015, Google, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START bookshelf_firestore_client]
+require "google/cloud/firestore"
+# [END bookshelf_firestore_client]
+require "google/cloud/storage"
+
+class Book
+  # Add Active Model support.
+  # Provides constructor that takes a Hash of attribute values.
+  include ActiveModel::Model
+
+  # Add Active Model validation support to Book class.
+  include ActiveModel::Validations
+
+  validates :title, presence: true
+
+  attr_accessor :id, :title, :author, :published_on, :description, :image_url, :cover_image
+
+  # Return a Google::Cloud::Firestore::Dataset for the configured collection.
+  # The collection is used to create, read, update, and delete entity objects.
+  def self.collection
+    project_id = ENV["GOOGLE_CLOUD_PROJECT"]
+    raise "Set the GOOGLE_CLOUD_PROJECT environment variable" if project_id.nil?
+
+    # [START bookshelf_firestore_client]
+    firestore = Google::Cloud::Firestore.new project_id: project_id
+    @collection = firestore.col "books"
+    # [END bookshelf_firestore_client]
+  end
+
+  def self.storage_bucket
+    project_id = ENV["GOOGLE_CLOUD_PROJECT"]
+    raise "Set the GOOGLE_CLOUD_PROJECT environment variable" if project_id.nil?
+
+    @storage_bucket = begin
+      config = Rails.application.config.x.settings
+      # [START bookshelf_cloud_storage_client]
+      bucket_id = project_id + ".appspot.com"
```

Comment:
This is the bucket automatically created for an app engine app. Which I guess is okay (I think it's intended for use by the app engine app for whatever it wants) but this conflicts with the text of the tutorial which says you have to create a bucket explicitly, named `${PROJECT_ID}_bucket`.